src/main/java/com/gregtechceu/gtceu/utils/GTUtil.java

```diff
@@ -396,6 +398,30 @@ public static FluidStack getFluidFromContainer(Object ingredient) {
         return null;
     }
 
+    /**
+     * Checks whether a machine is not a multiblock and has a recipemap not present in a blacklist
+     *
+     * @param machineStack the ItemStack containing the machine to check the validity of
+     * @return whether the machine is valid or not
+     */
+    public static boolean isMachineValidForMachineHatch(ItemStack machineStack, GTRecipeType[] recipeTypeBlacklist) {
+        if (machineStack == null || machineStack.isEmpty()) {
+            return false;
+        }
+
+        if (machineStack.getItem() instanceof MetaMachineItem metaMachineItem) {
+            MachineDefinition definition = metaMachineItem.getDefinition();
+            IMachineBlockEntity be = MetaMachineBlockEntity.createBlockEntity(definition.getBlockEntityType(), BlockPos.ZERO, definition.defaultBlockState());
+            MetaMachine machine = be.getMetaMachine();
+            if (machine instanceof WorkableTieredMachine && !(machine instanceof SimpleGeneratorMachine)) {
```

Comment:
I think it might be better to add an additional method for this check in the machine class instead, just so that addon devs can override that if necessary.  
Something like `isAllowedInProcessingArray()` maybe.

Not sure if this is fully covered by the recipe type blacklist already. Could have some edge cases here where a dummy recipe and custom recipe logic is used in addons.